pennylane/beta/tapes/tape.py

```diff
@@ -0,0 +1,776 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the base quantum tape.
+"""
+# pylint: disable=too-many-instance-attributes,protected-access,too-many-branches
+import contextlib
+
+import numpy as np
+
+import pennylane as qml
+
+from pennylane.beta.queuing import MeasurementProcess
+from pennylane.beta.queuing import AnnotatedQueue, QueuingContext
+from pennylane.beta.queuing import mock_operations
+
+from .circuit_graph import NewCircuitGraph
+
+
+STATE_PREP_OPS = (
```

Comment:
In the final PR of the series, we should move this to `pennylane/ops/qubit.py` etc. The definition of which ops are state preps should belong there, as part of the op definition.

Reply:
(TBD exactly how we do this. It could be a subclass of `Operation`, or it could simply be given in the module as a module variable)